databaseutil/databaseutil.go

```diff
@@ -60,36 +66,58 @@ func InsertIntoUsersTable(
 }
 
 func GetPasswordForUserWithEmailAddress(emailAddress string) ([]byte, error) {
-	var row *sql.Row
-	{
-		sqlQuery := `
-			SELECT password FROM users
-			WHERE email_address = $1`
+	sqlQuery := `
+		SELECT password FROM users
+		WHERE email_address = $1`
 
-		row = db.QueryRow(sqlQuery, emailAddress)
+	rows, err := db.Query(sqlQuery, emailAddress)
+	if err != nil {
+		return nil, convertPostgresError(err)
 	}
+	defer rows.Close()
 
 	var password []byte
-	if err := row.Scan(&password); err != nil {
-		return nil, err
+	for rows.Next() {
+		if password != nil {
+			return nil, QueryResultContainedMultipleRowsError
+		}
+
+		if err := rows.Scan(&password); err != nil {
+			return nil, err
+		}
+	}
+
+	if password == nil {
+		return nil, QueryResultContainedNoRowsError
 	}
 
 	return password, nil
 }
 
 func GetIdForUserWithEmailAddress(emailAddress string) (int64, error) {
-	var row *sql.Row
-	{
-		sqlQuery := `
-			SELECT id FROM users
-			WHERE email_address = $1`
+	sqlQuery := `
+		SELECT id FROM users
+		WHERE email_address = $1`
 
-		row = db.QueryRow(sqlQuery, emailAddress)
+	rows, err := db.Query(sqlQuery, emailAddress)
+	if err != nil {
+		return 0, convertPostgresError(err)
 	}
+	defer rows.Close()
 
 	var userId int64
-	if err := row.Scan(&userId); err != nil {
-		return 0, err
+	for rows.Next() {
```

Comment:
The next time we do this exact logic we should make it its own separate function.